web_extension/shared/options.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+class Options {
+  constructor() {
+    this.version = 1
+    this.display_options = (new DisplayOptions(this))
+    this.feed_options = (new FeedOptions(this))
+
+    this.fetched = false
+  }
+
+  async read () {
+    let promise = browser.storage.sync.get()
```

Comment:
```suggestion
    const stored_options = await browser.storage.sync.get()
    
    [...]
    
    return stored_options
```

but also is it `.sync` if its returning a promise?

Reply:
I did it this way because I want to return the promise up the chain, so I can await it elsewhere. I had a reason.

Reply:
And the deceptively named .sync is for [configuration which is sync'd by the ~firefox~ browser](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage#Properties)